### PR TITLE
LLM: optimize QLoRA by updating lora convert logic

### PR DESCRIPTION
**Pull request by rnwang04**
### 1. Why the change?

https://github.com/analytics-zoo/nano/issues/713#issuecomment-1798203085

### 2. User API changes

No change.

### 3. Summary of the change 

update convert logic of lora part to reduce convert time.

### 4. How to test?
- [x] Unit test
- [x] Local test

### CODE REVIEW

**Comment by rnwang04** [comment]
@jason-dai  As you said, putting the autocast at the first of LoraLowBitLinear does further improve the performance.
I have refactor related code, would you mind reviewing it again?

File: python/llm/src/bigdl/llm/transformers/utils.py [hunk 1]
```diff
@@ -133,3 +133,23 @@ def fix_key(key):
...
+
+
+def get_autocast_dtype(x):
+    if x.device.type == "xpu":
+        return torch.xpu.get_autocast_xpu_dtype()
+    elif x.device.type == "cpu":
+        return torch.get_autocast_cpu_dtype()
+    else:
+        invalidInputError(False,
+                          f"Device {x.device} is not supported.")
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/utils.py:155]: We can combine these functions into to one (e.g., `get_autocast_dtype()` returning `None` if not enabled) to reduce one conditional branch if needed

**Review by jason-dai — APPROVED** [review]
LGTM

File: python/llm/src/bigdl/llm/transformers/utils.py (hunk 1 above)
Comment (rnwang04) [reply python/llm/src/bigdl/llm/transformers/utils.py]: Have revised to this way.

**Comment by rnwang04** [comment]
@hzjane I will merge this PR first, please help to check whether this PR will effect CPU QLoRA tmr : )

**Pull request merged**